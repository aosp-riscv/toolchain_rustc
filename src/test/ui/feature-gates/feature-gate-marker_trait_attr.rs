use std::fmt::{Debug, Display};

#[marker] trait ExplicitMarker {}
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
//~^ ERROR marker traits is an experimental feature
=======
//~^ ERROR the `#[marker]` attribute is an experimental feature
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)

impl<T: Display> ExplicitMarker for T {}
impl<T: Debug> ExplicitMarker for T {}

fn main() {}
