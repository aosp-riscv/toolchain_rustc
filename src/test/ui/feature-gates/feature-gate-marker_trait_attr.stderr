<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error[E0658]: marker traits is an experimental feature
=======
error[E0658]: the `#[marker]` attribute is an experimental feature
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
  --> $DIR/feature-gate-marker_trait_attr.rs:3:1
   |
LL | #[marker] trait ExplicitMarker {}
   | ^^^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/29864
   = help: add `#![feature(marker_trait_attr)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
